### DISCUSSION TITLE
Flet Build APK Issue in New Release

### DISCUSSION THREAD

**Discussion by Muddassir-Farooq-official** [flet-dev/flet#4179] [Q&A]
### Duplicate Check

- [X] I have searched the [opened issues](https://github.com/flet-dev/flet/issues) and there are no duplicates

### Describe the bug

The bug is caused by a missing or incorrectly configured deep_linking attribute in the Flet project build process. Specifically, the build template you're using (flet-build-template) is trying to access a deep_linking configuration, but the attribute is either absent or improperly defined in the context of the project configuration.

The error occurs during the generation of the AndroidManifest.xml file, where deep linking permissions or configurations are required. The build system encounters an issue because it expects a deep_linking object in the context but can't find it, resulting in the following error:

'dict object' has no attribute 'deep_linking'

### Code sample

<details open><summary>Code</summary>

```python
import flet as ft


def main(page: ft.Page):
    page.add(ft.SafeArea(ft.Text("Hello, Flet!")))


ft.app(main)

```

</details>

### To reproduce

To reproduce the bug, follow these steps:

Prerequisites:
You need a system with latest pre release Flet installed.
Ensure you are using the Flet build template from flet-dev/flet-build-template.
You need to have Flutter, Android SDK, and Cookiecutter installed on your system.

### Screenshots / Videos

<details open>
<summary>Captures</summary>

[Upload media here]

</details>

### Operating System

Windows

### Operating system details

Windows 10

### Flet version

0.25.0.dev3519

### Regression

No, it isn't

### Logs

<details open><summary>Logs</summary>

```console
PS C:\Users\Muddassir Farooq\Desktop\Bonds> flet build apk
[09:51:35] Unable to create file 'android\app\src\main\AndroidManifest.xml'. Error message: 'dict object' has no attribute 'deep_linking'. Context: OrderedDict({'cookiecutter': OrderedDict({'out_dir': 'flutter',           'python_module_name': 'main', 'project_name': 'bonds', 'project_name_slug': '', 'project_description': '', 'product_name': 'bonds', 'org_name': 'com.flet', 'company_name': 'Your Company', 'copyright':           'Copyright (c) 2023 Your Company', 'flutter': {'dependencies': []}, 'sep': '\\', 'kotlin_dir': 'com\\flet\\bonds\\', 'hide_loading_animation': True, 'team_id': '', 'base_url': '/',
           'route_url_strategy': 'path', 'web_renderer': 'canvaskit', 'use_color_emoji': 'false', 'split_per_abi': False, 'options': {'info_plist': {}, 'macos_entitlements': {'com.apple.security.app-sandbox':   
           False, 'com.apple.security.cs.allow-jit': True, 'com.apple.security.network.client': True, 'com.apple.security.network.server': True}, 'android_permissions': {'android.permission.INTERNET': True},    
           'android_features': {'android.software.leanback': False, 'android.hardware.touchscreen': False}, 'deep_linking_url': None, 'android_signing': False}, '_template': 'gh:flet-dev/flet-build-template',   
           '_output_dir': 'C:\\Users\\Muddassir Farooq\\Desktop\\Bonds\\build', '_repo_dir': 'C:\\Users\\Muddassir Farooq\\.cookiecutters\\flet-build-template', '_checkout': '0.25.0'}), '_cookiecutter':
           {'out_dir': 'flutter', 'python_module_name': 'main', 'project_name': 'bonds', 'project_name_slug': '', 'project_description': '', 'product_name': 'bonds', 'org_name': 'com.flet', 'company_name': 'Your           Company', 'copyright': 'Copyright (c) 2023 Your Company', 'flutter': {'dependencies': []}, 'sep': '\\', 'kotlin_dir': "{{ cookiecutter.org_name.replace('.', cookiecutter.sep) }}{{ cookiecutter.sep    
           }}{{ cookiecutter.project_name }}{{ cookiecutter.sep }}", 'hide_loading_animation': True, 'team_id': '', 'base_url': '/', 'route_url_strategy': 'path', 'web_renderer': 'canvaskit', 'use_color_emoji': 
           'false', 'split_per_abi': False, 'options': {'info_plist': {}, 'macos_entitlements': {'com.apple.security.app-sandbox': False, 'com.apple.security.cs.allow-jit': True,
           'com.apple.security.network.client': True, 'com.apple.security.network.server': True}, 'android_permissions': {'android.permission.INTERNET': True}, 'android_features': {'android.software.leanback':  
           False, 'android.hardware.touchscreen': False}, 'deep_linking_url': None, 'android_signing': False}}})
[09:51:40] Doctor summary (to see all details, run flutter doctor -v):
           [√] Flutter (Channel stable, 3.24.1, on Microsoft Windows [Version 10.0.19045.5011], locale en-US)
           [√] Windows Version (Installed version of Windows is version 10 or higher)
           [√] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
           [√] Chrome - develop for the web
           [√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.10.0)
           [√] Android Studio (version 2023.2)
           [√] VS Code (version 1.94.2)
           [√] Connected device (3 available)
           [√] Network resources

           • No issues found!
```

</details>

**Comment by Muddassir-Farooq-official** [comment]
pyproject.toml 

[project]

name = "app"
version = "1.0.0"
description = ""

authors = [
  {name = "Flet Developer", email = "your@email.com"}
]

dependencies = ["flet==0.25.0.dev3526"]

[tool.flet]
permissions = ["camera", "microphone"] # --permissions

[tool.flet.android.permission] # --android-permissions
"android.permission.CAMERA" = true

[tool.flet.deep_linking]
scheme = "https" # --deep-linking-scheme
host = "mydomain.com" # --deep-linking-host

[tool.flet.android.deep_linking]
scheme = "https" # --deep-linking-scheme
host = "mydomain.com" # --deep-linking-host

[tool.flet.splash]
color = "pink" # --splash-color
dark_color = "pink" # --splash-dark-color
web = false # --no-web-splash
ios = false # --no-ios-splash
android = false # --no-android-splash

**Comment by FeodorFitsner** [comment]
Update Flet to 0.25.0.dev3526 https://flet.dev/blog/pyproject-toml-support-for-flet-build-command

**Comment by Muddassir-Farooq-official** [comment]
I tried it but I think there is a issue in deep linking

**Comment by FeodorFitsner** [comment]
OK, could you give me a log with `flet build apk -v`?

**Comment by Muddassir-Farooq-official** [comment]
Okay sure one thing I want to mention here kindly put documentation on new modules introduced in flet pre release like websockets and pillow

**Comment by Muddassir-Farooq-official** [comment]
> flet build apk -v

`PS C:\Users\Muddassir Farooq\Desktop\Bonds> flet build apk -v
[22:50:07] No additional Flutter dependencies!
[22:50:10] Unable to create file 'android\app\src\main\AndroidManifest.xml'. Error message: 'dict object' has no attribute 'deep_linking'. Context: OrderedDict({'cookiecutter': OrderedDict({'out_dir': 'flutter',           'python_module_name': 'main', 'project_name': 'bonds', 'project_name_slug': '', 'project_description': '', 'product_name': 'bonds', 'org_name': 'com.flet', 'company_name': 'Your Company', 'copyright':           'Copyright (c) 2023 Your Company', 'flutter': {'dependencies': []}, 'sep': '\\', 'kotlin_dir': 'com\\flet\\bonds\\', 'hide_loading_animation': True, 'team_id': '', 'base_url': '/',
           'route_url_strategy': 'path', 'web_renderer': 'canvaskit', 'use_color_emoji': 'false', 'split_per_abi': False, 'options': {'info_plist': {}, 'macos_entitlements': {'com.apple.security.app-sandbox':   
           False, 'com.apple.security.cs.allow-jit': True, 'com.apple.security.network.client': True, 'com.apple.security.network.server': True}, 'android_permissions': {'android.permission.INTERNET': True},    
           'android_features': {'android.software.leanback': False, 'android.hardware.touchscreen': False}, 'deep_linking_url': None, 'android_signing': False}, '_template': 'gh:flet-dev/flet-build-template',   
           '_output_dir': 'C:\\Users\\Muddassir Farooq\\Desktop\\Bonds\\build', '_repo_dir': 'C:\\Users\\Muddassir Farooq\\.cookiecutters\\flet-build-template', '_checkout': '0.25.0'}), '_cookiecutter':
           {'out_dir': 'flutter', 'python_module_name': 'main', 'project_name': 'bonds', 'project_name_slug': '', 'project_description': '', 'product_name': 'bonds', 'org_name': 'com.flet', 'company_name': 'Your           Company', 'copyright': 'Copyright (c) 2023 Your Company', 'flutter': {'dependencies': []}, 'sep': '\\', 'kotlin_dir': "{{ cookiecutter.org_name.replace('.', cookiecutter.sep) }}{{ cookiecutter.sep    
           }}{{ cookiecutter.project_name }}{{ cookiecutter.sep }}", 'hide_loading_animation': True, 'team_id': '', 'base_url': '/', 'route_url_strategy': 'path', 'web_renderer': 'canvaskit', 'use_color_emoji': 
           'false', 'split_per_abi': False, 'options': {'info_plist': {}, 'macos_entitlements': {'com.apple.security.app-sandbox': False, 'com.apple.security.cs.allow-jit': True,
           'com.apple.security.network.client': True, 'com.apple.security.network.server': True}, 'android_permissions': {'android.permission.INTERNET': True}, 'android_features': {'android.software.leanback':  
           False, 'android.hardware.touchscreen': False}, 'deep_linking_url': None, 'android_signing': False}}})
           Run subprocess: ['C:\\src\\flutter_windows_3.19.6-stable\\flutter\\bin\\flutter.BAT', 'doctor']
[22:50:16] Doctor summary (to see all details, run flutter doctor -v):
           [√] Flutter (Channel stable, 3.24.1, on Microsoft Windows [Version 10.0.19045.5011], locale en-US)
           [√] Windows Version (Installed version of Windows is version 10 or higher)
           [√] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
           [√] Chrome - develop for the web
           [√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.10.0)
           [√] Android Studio (version 2023.2)
           [√] VS Code (version 1.94.2)
           [√] Connected device (3 available)
           [√] Network resources

           • No issues found!`

**Comment by FeodorFitsner** [comment]
What you get with `flet --version`?

**Comment by Muddassir-Farooq-official** [comment]
> What you get with `flet --version`?

0.25.0.dev3519

**Comment by FeodorFitsner** [comment]
Upgrade Flet to 3526:

```
pip install flet==0.25.0.dev3526 --upgrade
```

**Comment by Muddassir-Farooq-official** [comment]
> pip install flet==0.25.0.dev3526 --upgrade

`PS C:\Users\Muddassir Farooq\Desktop\Bonds> flet build apk --verbose
[09:48:30] No additional Flutter dependencies!
[09:48:34] Created Flutter bootstrap project from gh:flet-dev/flet-build-template with ref 0.25.0 ✅                                                                                                               
           Copying C:\Users\Muddassir Farooq\Desktop\Bonds\assets\icon.png to C:\Users\Muddassir Farooq\Desktop\Bonds\build\flutter\images
           Customized app icons and splash images ✅                                                                                                                                                               
           Run subprocess: ['C:\\src\\flutter_windows_3.19.6-stable\\flutter\\bin\\dart.BAT', 'run', 'flutter_launcher_icons']
( ●    ) Generating app icons ⏳... 
( ●    ) Generating app icon(10.3s) 
                                                                            ( ●    ) Generating app icons ⏳... 
( ●    ) Generating app icons ⏳... 
Built flutter_launcher_icons:flutter_launcher_icons.
(     ●) Generating app icons ⏳...   ════════════════════════════════════════════
     FLUTTER LAUNCHER ICONS (v0.14.1)
  ════════════════════════════════════════════

(    ● ) Generating app icons ⏳...• Creating default icons Android
(   ●  ) Generating app icons ⏳...• Overwriting the default Android launcher icon with a new icon
( ●    ) Generating app icons ⏳...• Creating mipmap xml file Android
(  ●   ) Generating app icons ⏳...• Overwriting default iOS launcher icon with new icon
( ●    ) Generating app icons ⏳... Creating Icons for Web...
(     ●) Generating app icons ⏳...Creating Icons for Windows...
(  ●   ) Generating app icons ⏳...Creating Icons for MacOS...
(     ●) Generating app icons ⏳... 
✓ Successfully generated launcher icons
[09:48:59] Generated app icons ✅                                                                                                                                                                                  
           Run subprocess: ['C:\\src\\flutter_windows_3.19.6-stable\\flutter\\bin\\dart.BAT', 'run', 'flutter_native_splash:create']
(    ● ) Generating splash screens(2.6s) 
(   ●  ) Generating splash screens ⏳... Built flutter_native_splash:create.
(   ●  ) Generating splash screens ⏳...Unhandled exception:
Exception: Invalid color value
#0      parseColor (package:flutter_native_splash/cli_commands.dart:416:3)
#1      _createSplashByConfig (package:flutter_native_splash/cli_commands.dart:87:25)
#2      createSplash (package:flutter_native_splash/cli_commands.dart:43:3)
#3      main (file:///C:/Users/Muddassir%20Farooq/AppData/Local/Pub/Cache/hosted/pub.dev/flutter_native_splash-2.4.1/bin/create.dart:31:5)
#4      _delayEntrypointInvocation.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:295:33)
#5      _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:184:12)
[09:49:03] Error building Flet app - see the log of failed command above.
           Run subprocess: ['C:\\src\\flutter_windows_3.19.6-stable\\flutter\\bin\\flutter.BAT', '--version']
[09:49:06] Run subprocess: ['C:\\src\\flutter_windows_3.19.6-stable\\flutter\\bin\\flutter.BAT', 'doctor']
[09:49:11] Doctor summary (to see all details, run flutter doctor -v):
           [√] Flutter (Channel stable, 3.24.1, on Microsoft Windows [Version 10.0.19045.5011], locale en-US)
           [√] Windows Version (Installed version of Windows is version 10 or higher)
           [√] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
           [√] Chrome - develop for the web
           [√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.10.0)
           [√] Android Studio (version 2023.2)
           [√] VS Code (version 1.94.2)
           [√] Connected device (3 available)
           [√] Network resources

           • No issues found!`

**Comment by FeodorFitsner** [comment]
```
...
( ● ) Generating splash screens ⏳...Unhandled exception:
Exception: Invalid color value
...
```

**Comment by FeodorFitsner** [comment]
Try hex color instead of `pink`.

**Comment by Muddassir-Farooq-official** [comment]
how can we integrate notification permission and send notification in flet latest release

**Comment by Muddassir-Farooq-official** [comment]
> Try hex color instead of `pink`.

after running build apk its generating this all things and when i open apk on mobile its showing black screen nothing happens next
![image](https://github.com/user-attachments/assets/f4ee4e65-7d41-4b09-8e32-231fb24651ef)

**Comment by FeodorFitsner** [comment]
What Android device do you have? Try running on Android simulator?

**Comment by Muddassir-Farooq-official** [comment]
> What Android device do you have? Try running on Android simulator?

I am using Techno Camon 19 Neo

**Reply by Muddassir-Farooq-official** [reply]
First app shows blank screen then showing unknown control webview but I didn't use webview in my code

**Comment by Muddassir-Farooq-official** [comment]
> What Android device do you have? Try running on Android simulator?

https://github.com/user-attachments/assets/add9324e-17d9-4ebe-be6a-7bde6df00acf

**Reply by FeodorFitsner** [reply]
Honestly, it's impossible to tell whether your app is using WebView or not without looking at the sources, right? From "buildability" standpoint it looks like the app is running - `Unknown control` couldn't be displayed by its own. Maybe it's an old build, idk.

**Reply by Muddassir-Farooq-official** [reply]
check it out https://github.com/flet-dev/flet/issues/4386
